IoTEnterpriseDefault/CS/SmartDisplay.Utils/Authentication/WamProvider.cs

```diff
@@ -0,0 +1,362 @@
+                {
+                    if (!string.IsNullOrWhiteSpace(data.Token))
+                    {
+                        return data.Token;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                ServiceUtil.LogService.Write(ex.ToString(), LoggingLevel.Warning);
+            }
+
+            return null;
+        }
+
+        private WebAccountProvider _wap;
+        private async Task<WebAccountProvider> GetProvider()
+        {
+            if (_wap == null)
+            {
+                _wap = await WebAuthenticationCoreManager.FindAccountProviderAsync(ProviderId, _authority);
+            }
+            return _wap;
+        }
+
+        private string GetSavedAccountId()
+        {
+            return _appSettings.Values[_userIdKey] as string;
+        }
+
+        private void ClearSavedAccountId()
+        {
+            _appSettings.Values[_userIdKey] = null;
+        }
+
+        private bool SaveToken(WebTokenRequestResult wtrr, string resource)
+        {
+            if (wtrr.ResponseStatus == WebTokenRequestStatus.Success)
+            {
+                if (resource == null)
+                {
+                    resource = DefaultResource;
+                }
+
+                if (!_tokens.ContainsKey(resource))
+                {
+                    _tokens.Add(resource, new TokenInfo());
+                }
+
+                _tokens[resource].Token = GetTokenFromResult(wtrr);
+                CachedProperties.Clear();
+
+                if (wtrr.ResponseData != null)
+                {
+                    foreach (var data in wtrr.ResponseData)
+                    {
+                        CachedProperties = new Dictionary<string, string>(data.Properties);
+
+                        if (data.Properties.TryGetValue("exp", out string unixTimeString))
+                        {
+                            var dt = DateTimeOffset.FromUnixTimeSeconds(Convert.ToUInt32(unixTimeString));
+                            _tokens[resource].ExpiresOn = dt.DateTime.ToLocalTime();
+                            ServiceUtil.LogService.Write($"Resource: {resource}, Token Expiration: {_tokens[resource].ExpiresOn}");
+                        }
+
+                        _isSignedIn = true;
+                        TokenUpdate?.Invoke(this, new EventArgs());
+                        return true;
+                    }
+                }
+            }
+
+            _isSignedIn = false;
+            return false;
+        }
+
+        private void ClearTokens()
+        {
+            _tokens.Clear();
+            CachedProperties.Clear();
+        }
+
+        private async Task<WebTokenRequestResult> RequestTokenWithTimeout(WebTokenRequest request)
+        {
+            // The WebTokenRequest will time out if the user does not complete in time.
+            // This is because there is currently no 'close' button on IoT Enterprise, so it
```

Comment:
There is a 'close' button on IoT Enterprise.  Please change comment back to IoT Core which this comment is applicable to.

Reply:
Made the changes